File: tree/postOrder.py
class Node:
    def __init__(self, val):
        self.data = val
        self.left = None
        self.right = None

def postOrder(root, arr):
    if not root:
        return
    postOrder(root.left, arr)
    postOrder(root.right, arr)
    arr.append(root.data)


root = Node(1)
root.left = Node(2)
root.right = Node(3)
root.left.left = Node(4)
root.left.right = Node(5)

result = []
postOrder(root, result)

print("postOrder Traversal:", end=" ")
for val in result:
    print(val, end=" ")
print()

File: tree/postOrder.cpp
#include <iostream>
#include <vector>
using namespace std;

// Define Node class
class Node {
public:
    int data;
    Node* left;
    Node* right;

    // Constructor
    Node(int val) {
        data = val;
        left = right = nullptr;
    }
};

// Postorder traversal function
// left right root
void postOrder(Node* root, vector<int>& arr) {
    if (root == nullptr) return;

    postOrder(root->left, arr);
    postOrder(root->right, arr);
    arr.push_back(root->data);
}

int main() {
    // Build the tree
    /*
            1
           / \
          2   3
         / \
        4   5
    */
    Node* root = new Node(1);
    root->left = new Node(2);
    root->right = new Node(3);
    root->left->left = new Node(4);
    root->left->right = new Node(5);

    // Store traversal result
    vector<int> result;
    postOrder(root, result);

    // Print result
    cout << "Postorder Traversal: ";
    for (int val : result) {
        cout << val << " ";
    }
    cout << endl;

    return 0;
}
